.github/workflows/sonarcloud.yaml

```diff
@@ -0,0 +1,27 @@
+on:
+  # Trigger analysis when pushing in master or pull requests, and when creating
+  # a pull request.
+  push:
+    branches:
+      - master
+  pull_request:
+      types: [opened, synchronize, reopened]
+name: Sonarcloud Workflow
+jobs:
+  sonarcloud:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+    - name: setup dev env
+      uses: ./.github/actions/dev-env-setup
+    - name: Run Jest tests
+      working-directory: ./app
+      run: yarn test --coverage
```

Comment:
We should update the `jest` job in the tests workflow, rather than adding another workflow that runs `yarn test` ?

Reply:
I'm probably going about this the wrong way, but when attempting to reuse the jests workflow the coverage data is wiped on the new checkout, and it isn't persisted between jobs. Any guidance on this?

Reply:
Changed the jest tests to an action and reusing that, let me know if that's sufficient